mitodl/micromasters
# Handle 503 error on dashboard api

#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/3927

#### What's this PR do?
- When there is 503 error on dashboard api fetchJSONWithCSRF expects json but we have text/html error message. 
- And in this case of error we are returning success promise instead of rejection
- because of that dashboard neither have content nor the error info
- I fixed this by returning `Promise.reject('Unable to process')

#### How should this be manually tested?
- go to nginx.conf and put these lines to create 503 error
```
    location /api/v0/dashboard/ {
        include uwsgi_params;
        if ($request_method = GET ) {
            return 503;
        }
    }
```
- And on dashboard you should see error message
- you may need to restart docker

@pdpinch 

#### Screenshots (if appropriate)
<img width="1008" alt="screen shot 2018-04-23 at 8 17 07 pm" src="https://user-images.githubusercontent.com/10431250/39136335-6ebfdec4-4734-11e8-9cc4-d5b458411a72.png">
